projects/dsp-ui/src/lib/search/fulltext-search/fulltext-search.component.spec.ts

```diff
@@ -0,0 +1,399 @@
+        testHostFixture.detectChanges();
+
+        // check the local storage state
+        expect(testHostComponent.fulltextSearch.searchQuery).toEqual('new thing');
+        const newPrevSearchArray: PrevSearchItem[] = [
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'one thing' },
+            { query: 'new thing' }
+        ];
+        expect(localStorage.getItem('prevSearch')).toEqual(JSON.stringify(newPrevSearchArray));
+
+        // check the call of router.navigate with correct arguments
+        expect(mockRouter.navigate).toHaveBeenCalledWith(['/search/fulltext/new thing']);
+
+    });
+
+    // todo: fix error: Cannot read property 'detach' of undefined CDK OVERLAY
+    /* xit('should reset the current search', () => {
+        const searchInput = fulltextSearchComponentDe.query(By.css('input.kui-fulltext-search-input')).nativeElement;
+        expect(searchInput).toBeDefined();
+
+        searchInput.value = 'thing to search';
+        expect(searchInput.value).toEqual('thing to search');
+
+        testHostFixture.detectChanges();
+
+        // searchInput.dispatchEvent(new KeyboardEvent('keyup', { key: 'Esc' }));
+        testHostComponent.fulltextSearch.resetSearch();
+
+        expect(searchInput.value).toEqual('');
+    }); */
+
+    it('should perform a search with a previous item - solution 1', () => {
+        const prevSearchArray: PrevSearchItem[] = [
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'blue thing' },
+            { projectIri: 'http://rdfh.ch/projects/0801', projectLabel: 'anything', query: 'purple thing' }
+        ];
+
+        localStorage.setItem('prevSearch', JSON.stringify(prevSearchArray));
+        expect(localStorage.getItem('prevSearch')).toBeDefined();
+
+        // click in the search input to open the search panel
+        const searchInput = fulltextSearchComponentDe.query(By.css('input.kui-fulltext-search-input')).nativeElement;
+        expect(searchInput).toBeDefined();
+        searchInput.click();
+        testHostFixture.detectChanges();
+
+        const searchMenuPanel = fulltextSearchComponentDe.query(By.css('div.kui-search-menu')).nativeElement;
+        expect(searchMenuPanel).toBeDefined();
+
+        const prevSearchItem = fulltextSearchComponentDe.query(By.css('div.kui-previous-search-query')).nativeElement;
+        prevSearchItem.click();
+        testHostFixture.detectChanges();
+
+        expect(testHostComponent.fulltextSearch.searchQuery).toEqual('purple thing');
+        expect(testHostComponent.fulltextSearch.projectIri).toEqual('http://rdfh.ch/projects/0801');
+        expect(testHostComponent.fulltextSearch.projectLabel).toEqual('anything');
+
+        expect(mockRouter.navigate).toHaveBeenCalledWith(['/search/fulltext/purple thing/http%3A%2F%2Frdfh.ch%2Fprojects%2F0801']);
+
+    });
+
+    // todo: fix error: Cannot read property 'detach' of undefined CDK OVERLAY
```

Comment:
@kilchenmann @tobiasschweizer same comment sa above: I get the following error: Cannot read property 'detach' of undefined. I suppose I have to define Overlay somewhere but I looked everywhere and tried several tutorials unsuccessfully.... Any idea?